lib/redis_mutex/supervisor.ex

```diff
@@ -0,0 +1,22 @@
+defmodule RedisMutex.Supervisor do
+  @moduledoc """
+  The supervisor module for RedisMutex.
+  """
+  use Supervisor
+
+  @spec start_link(module(), RedisMutex.start_options()) :: Supervisor.on_start()
+  def start_link(lock_module, opts) when is_atom(lock_module) and is_list(opts) do
+    Supervisor.start_link(__MODULE__, {lock_module, opts}, name: __MODULE__)
+  end
```

Comment:
It feels weird to me that there are so many child_spec/start_link things wrapping each other.

Reply:
Would it make more sense to have the top-level module, `RedisMutex` be the supervisor for the lock module? In other words, just eliminate `RedisMutex.Supervisor`.

Reply:
yea I agree with @brettbeatty , Lets  make the API simple and underlying delegate to what is needed .